eureka2-client/src/main/java/com/netflix/eureka2/client/resolver/EurekaServerResolver.java

```diff
@@ -219,14 +250,20 @@ public EurekaServerResolver build() {
             if (transportConfig == null) {
                 transportConfig = new BasicEurekaTransportConfig.Builder().build();
             }
-            if(metricFactory == null) {
-                metricFactory = EurekaClientMetricFactory.clientMetrics();
+            if (clientMetricFactory == null) {
+                clientMetricFactory = EurekaClientMetricFactory.clientMetrics();
+            }
+            if (registryMetricFactory == null) {
+                registryMetricFactory = EurekaRegistryMetricFactory.registryMetrics();
             }
 
-            TcpDiscoveryClient transportClient =
-                    (TcpDiscoveryClient) TransportClients.newTcpDiscoveryClient(transportConfig, bootstrapResolver,metricFactory);
-            SnapshotInterestChannel snapshotInterestChannel = new SnapshotInterestChannelImpl(transportClient);
-            return new EurekaServerResolver(snapshotInterestChannel, readServerInterest, serviceSelector, loadBalancerBuilder);
+            EurekaClientBuilder eurekaClientBuilder = EurekaClientBuilder.newBuilder()
+                    .withMetricFactory(clientMetricFactory)
+                    .withMetricFactory(registryMetricFactory)
+                    .withWriteServerResolver(ServerResolvers.just("localhost", 0)) // TODO We have provide some resolver, otherwise validation fails
```

Comment:
At part of the client reorganisation, the final EurekaClient should be provided as a wrapper of both a RegistrationClient and an InterestClient. When that is available, we can just use a raw InterestClient here.